legacy/ExportPIM.js

```diff
@@ -19,7 +19,7 @@ async function LegacyExportPIM(req) {
     console.log('error: ', err);
   }
   
-  if (recordsAndCols.length !== 2) {
+  if (recordsAndCols == null || recordsAndCols.length !== 2) {
```

Comment:
I like the early exit, but let's also adopt js features (optional chaining) with `?.`

`recordsAndCols?.length`